src/lib/modules/scaffolding/index.ts

```diff
@@ -0,0 +1,60 @@
+import { Contract } from "../../../typings/contract";
+import { Embark } from "../../../typings/embark";
+import { Solidity } from "./contractLanguage/solidity";
+import { Description } from "./description";
+import { React } from "./framework/react";
+import { ContractLanguage, Framework, Options } from "./options";
+
+export default class Scaffolding {
+
+  constructor(private embark: Embark, private options: any) {
+    this.embark.events.setCommandHandler("scaffolding:generate:contract", this.generateContract.bind(this));
+    this.embark.events.setCommandHandler("scaffolding:generate:ui", this.generateUi.bind(this));
+  }
+
+  private readonly contractLanguageStrategy = (description: Description, options: Options) => {
+    switch (options.contractLanguage) {
+      case ContractLanguage.Solidity: {
+        return new Solidity(this.embark, description, options);
+      }
```

Comment:
Hmm, I just saw you do the validate in the Options.
I still feel like it should be this functions responsibility to do it, because it is now guaranteed that this function will be called after validate.
What you could do is move the validate inside this function.

Reply:
I am using advantage of typescript here.
If I don't have all case specified, TS checker will tell me to add it.

It is then a matter of responsability.
1 - Parse the option and validated
=> acts as a firewall and protect all the code after.
2 - Do the logic
=>  I am sure that the data is valid

Why adding the validate here? and why not inside each function using the option then?
For example I also validate `overwrite`. That would mean I would have to call validate before using overwrite too if I follow your logic?

If you look at the Solidity/React code there is no validation there because my options has been validated already

However, I could call validate when I build the object, would that satisfy you?

Reply:
I understand. It's fine if you leave it like this then